xds/src/main/java/io/grpc/xds/internal/security/certprovider/CertProviderClientSslContextProvider.java

```diff
@@ -46,7 +44,7 @@ final class CertProviderClientSslContextProvider extends CertProviderSslContextP
         node,
         certProviders,
         certInstance,
-        checkNotNull(rootCertInstance, "Client SSL requires rootCertInstance"),
```

Comment:
There are usages of isMtls() /isClientSideTls()/isServerSideTls() for updating the SslContext. 
The [usage](https://github.com/grpc/grpc-java/blob/master/xds/src/main/java/io/grpc/xds/internal/security/certprovider/CertProviderServerSslContextProvider.java#L64) if isMtls() in server cert provider for setting the trust manager with the ca root certs won't be impacted because the change is only on the client side. In any case I think if (savedTrustedRoots != null) would do the same work as the isMtls() check here.

The [usage](https://github.com/grpc/grpc-java/blob/master/xds/src/main/java/io/grpc/xds/internal/security/certprovider/CertProviderClientSslContextProvider.java#L65) of isMtls() in client cert provider for setting the keyManager is impacted by the change, and is rather done better with the check if (savedKey != null) instead of if (isMtls()).

The [usages in the base class CertProviderSslContextProvider](https://github.com/grpc/grpc-java/blob/master/xds/src/main/java/io/grpc/xds/internal/security/certprovider/CertProviderSslContextProvider.java#L152) of isMtls() / isClientSideTls() / isServerSideTls() is very confusing to me, since both the client side and server side providers inherit from it (i.e. since a connection has a Xds client and a Xds server, will calling any of these methods on either party should return the same result, it doesn't look like it). Each of these usages also have a further nested check based on savedKey or savedTrustedRoots, I think the inner if conditions can just be used by themselves without the check first for sMtls() / isClientSideTls() / isServerSideTls(), unless those checks achieve anything meaningful.

Reply:
For the usages in the base class CertProviderSslContextProvider that I mentioned above can we use only the inner if conditions under each of the isMtls() / isClientSideTls() / isServerSideTls() ?

Reply:
> won't be impacted because the change is only on the client side

Well, that's assuming we notice the issue if/when we add server support.

It looks like we should get rid of `isMtls()` and instead have new methods to check if keys or cert validation is configured, to be used as necessary.

Reply:
I realized the trio of methods isMtls() / isClientSideTls() / isServerSideTls() help do 2 things - 
1. When either the key or trusted roots get updated, instead of just calling updateSslContext with that value, it is used to first check whether either or both of key / trusted roots as applicable is not null before calling updateSslContext.
2. And in the client side, if Mtls it will add client certificate to the context, and on the server side if Mtls it will add trusted roots to the context.

I think the easy way and the only thing needed is to augment isMtls() to 

```
protected final boolean isMtls() {
    return certInstance != null && (rootCertInstance != null || isUsingSystemRootCerts;
  }

```
I have also introduced a test for Mtls with system roots for the client: XdsSecurityClientServerTest::tlsClientServer_useSystemRootCerts_requireClientAuth.